src/index.js

```diff
@@ -50,37 +51,40 @@ LFS.prototype = {
   untrack,
 };
 
-module.exports = (
-  nodegit,
-  {
-    nodeBinaryPath = process.execPath,
-    defaultShellOptions = null,
-    // when built, askpass.sh will be two directories above index.js
-    gitAskPassPath = require.resolve('../../askpass.sh'),
-    gitAskPassClientPath = require.resolve('./utils/GitAskPass')
-  } = {}
-) => {
-  const _NodeGit = nodegit; // eslint-disable-line no-underscore-dangle
+module.exports = {
+  initLfs: (
+    nodegit,
+    {
+      nodeBinaryPath = process.execPath,
+      defaultShellOptions = null,
+      // when built, askpass.sh will be two directories above index.js
+      gitAskPassPath = require.resolve('../../askpass.sh'),
+      gitAskPassClientPath = require.resolve('./utils/GitAskPass')
+    } = {}
+  ) => {
+    const _NodeGit = nodegit; // eslint-disable-line no-underscore-dangle
 
-  Object.getPrototypeOf(_NodeGit).LFS = new LFS(_NodeGit);
+    Object.getPrototypeOf(_NodeGit).LFS = new LFS(_NodeGit);
 
-  module.exports = _NodeGit;
+    module.exports = _NodeGit;
 
-  if (nodeBinaryPath) {
-    setNodeBinaryPath(nodeBinaryPath);
-  }
+    if (nodeBinaryPath) {
+      setNodeBinaryPath(nodeBinaryPath);
+    }
 
-  if (gitAskPassPath) {
-    setGitAskPassPath(gitAskPassPath);
-  }
+    if (gitAskPassPath) {
+      setGitAskPassPath(gitAskPassPath);
+    }
 
-  if (gitAskPassClientPath) {
-    setGitAskPassClientPath(gitAskPassClientPath);
-  }
+    if (gitAskPassClientPath) {
+      setGitAskPassClientPath(gitAskPassClientPath);
+    }
 
-  if (defaultShellOptions) {
-    setDefaultShellOptions(defaultShellOptions);
-  }
+    if (defaultShellOptions) {
+      setDefaultShellOptions(defaultShellOptions);
+    }
 
-  return _NodeGit;
+    return _NodeGit;
+  },
+  lfsDebug: LFS_DEBUG
```

Comment:
Let's leave the original export alone, but attach `lfsDebug` to `module.exports` like this:

```javascript
module.exports = () => {} /* original function here */
module.exports.lfsDebug = LFS_DEBUG;
```

Remember, functions are first class objects in Javascript, so we can assign the debug property to the main export.

Reply:
Great idea! Really cleans it up!